src/components/DatePicker/index.tsx

```diff
@@ -1,85 +1,86 @@
+import type {ForwardedRef} from 'react';
 import {View} from 'react-native';
 import * as Expensicons from '@components/Icon/Expensicons';
-import refPropTypes from '@components/refPropTypes';
 import TextInput from '@components/TextInput';
-import {propTypes as baseTextInputPropTypes, defaultProps as defaultBaseTextInputPropTypes} from '@components/TextInput/BaseTextInput/baseTextInputPropTypes';
+import type {BaseTextInputProps, BaseTextInputRef} from '@components/TextInput/BaseTextInput/types';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
 import CONST from '@src/CONST';
 import CalendarPicker from './CalendarPicker';
 
-const propTypes = {
-    /** React ref being forwarded to the DatePicker input */
-    forwardedRef: refPropTypes,
-
+type DatePickerProps = {
     /**
      * The datepicker supports any value that `new Date()` can parse.
      * `onInputChange` would always be called with a Date (or null)
      */
-    value: PropTypes.string,
+    value?: string;
 
     /**
      * The datepicker supports any defaultValue that `new Date()` can parse.
      * `onInputChange` would always be called with a Date (or null)
      */
-    defaultValue: PropTypes.string,
+    defaultValue?: string;
 
-    inputID: PropTypes.string.isRequired,
+    inputID: string;
 
     /** A minimum date of calendar to select */
-    minDate: PropTypes.objectOf(Date),
+    minDate?: Date;
 
     /** A maximum date of calendar to select */
-    maxDate: PropTypes.objectOf(Date),
+    maxDate?: Date;
 
     /** A function that is passed by FormWrapper */
-    onInputChange: PropTypes.func.isRequired,
+    onInputChange: (value: Date) => void;
 
     /** A function that is passed by FormWrapper */
-    onTouched: PropTypes.func.isRequired,
-
-    ...baseTextInputPropTypes,
-};
-
-const datePickerDefaultProps = {
-    ...defaultBaseTextInputPropTypes,
-    minDate: setYear(new Date(), CONST.CALENDAR_PICKER.MIN_YEAR),
-    maxDate: setYear(new Date(), CONST.CALENDAR_PICKER.MAX_YEAR),
-    value: undefined,
-};
-
-function DatePicker({forwardedRef, containerStyles, defaultValue, disabled, errorText, inputID, isSmallScreenWidth, label, maxDate, minDate, onInputChange, onTouched, placeholder, value}) {
+    onTouched: () => void;
+} & BaseTextInputProps;
+
+function DatePicker(
+    {
+        containerStyles,
+        defaultValue,
+        disabled,
+        errorText,
+        inputID,
+        label,
+        maxDate = setYear(new Date(), CONST.CALENDAR_PICKER.MAX_YEAR),
+        minDate = setYear(new Date(), CONST.CALENDAR_PICKER.MIN_YEAR),
+        onInputChange,
+        onTouched,
+        placeholder,
+        value,
+    }: DatePickerProps,
+    ref: ForwardedRef<BaseTextInputRef>,
+) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
```

Comment:
Sorry, mind explaining why we cant use nullish coalescing here?

Reply:
Here `value` and `defaultValue` can be an empty string, in this case we need to set the value to `undefined`. Nullish coalescing accepts an empty string so I replaced it with the`||` operator

Reply:
Got it!